brainglobe_atlasapi/atlas_generation/atlas_scripts/demba_dev_mouse.py

```diff
@@ -0,0 +1,345 @@
+
+
+def retrieve_additional_references(
+    download_dir_path, age, resolution, modalities
+):
+    """This function only needs editing if the atlas has additional reference
+    images. It should return a dictionary that maps the name of each
+    additional reference image to an image stack containing its data.
+    """
+    additional_references = {}
+    for modality in modalities:
+        reference_path, _, volume_resolution = (
+            get_reference_and_annotation_paths(
+                download_dir_path, age, modality
+            )
+        )
+        ref = load_any(reference_path)
+        zoom_factors = tuple(volume_resolution / resolution for _ in range(3))
+        ref = zoom(ref, zoom_factors, order=1)
+        additional_references[modality] = ref
+    return additional_references
+
+
+def retrieve_hemisphere_map():
+    """
+    Retrieve a hemisphere map for the atlas.
+
+    If your atlas is asymmetrical, you may want to use a hemisphere map.
+    This is an array in the same shape as your template,
+    with 0's marking the left hemisphere, and 1's marking the right.
+
+    If your atlas is symmetrical, ignore this function.
+
+    Returns:
+        numpy.array or None: A numpy array representing the hemisphere map,
+        or None if the atlas is symmetrical.
+    """
+    return None
+
+
+def retrieve_structure_information():
+    """
+    Retrieve the structures tree and meshes for the Allen mouse brain atlas.
+
+    Returns:
+        pandas.DataFrame: A DataFrame containing the atlas information.
+    """
+    # Since this atlas inherits from the allen can we not simply get the data
+    # from the bgapi?
+    print("determining structures")
+    allen_atlas = BrainGlobeAtlas("allen_mouse_25um")
+    allen_structures = allen_atlas.structures_list
+    allen_structures = [
+        {
+            "id": i["id"],
+            "name": i["name"],
+            "acronym": i["acronym"],
+            "structure_id_path": i["structure_id_path"],
+            "rgb_triplet": i["rgb_triplet"],
+        }
+        for i in allen_structures
+    ]
+    return allen_structures
+
+
+def retrieve_or_construct_meshes(
+    structures, annotated_volume, download_dir_path
+):
+    """
+    This function should return a dictionary of ids and corresponding paths to
+    mesh files. We construct the meshes ourselves for this atlas, as the
+    original data does not provide precomputed meshes.
+    """
+    print("constructing meshes")
+    meshes_dir_path = download_dir_path / "meshes"
+    meshes_dir_path.mkdir(exist_ok=True)
+    tree = get_structures_tree(structures)
+    labels = np.unique(annotated_volume).astype(np.int32)
+    for key, node in tree.nodes.items():
+        if key in labels:
+            is_label = True
+        else:
+            is_label = False
+
+        node.data = Region(is_label)
+
+    # Mesh creation
+    closing_n_iters = 2  # not used for this atlas
+    decimate_fraction = 0.2  # not used for this atlas
```

Comment:
```suggestion
    closing_n_iters = 2
    decimate_fraction = 0.2
```
Comments not true, I think?

Reply:
This code has been copy and pasted from one script to another like ten times now. It is both cursed and haunted by comments from atlases past.